src/utils/shaderUtils.ts

```diff
@@ -0,0 +1,161 @@
+import { Shader, ShaderProgram, ShaderTypes } from "../types";
+
+function createShader(
+  ctx: WebGL2RenderingContext,
+  type: ShaderTypes,
+  src: string
+) {
+  var shader = ctx.createShader(ctx[type]);
+
+  if (!shader) {
+    throw new Error("Failed to create shader.");
+  }
+
+  ctx.shaderSource(shader, src);
+  ctx.compileShader(shader);
+  var compiled = ctx.getShaderParameter(shader, ctx.COMPILE_STATUS);
+  if (!compiled) {
+    var lastError = ctx.getShaderInfoLog(shader);
+    ctx.deleteShader(shader);
+    throw new Error("*** Error compiling shader '" + shader + "':" + lastError);
+  }
+  return shader;
+}
+
+function createProgram(
+  ctx: WebGL2RenderingContext,
+  shader: Shader
+): WebGLProgram {
+  var vshader = createShader(ctx, "VERTEX_SHADER", shader.vertex);
+  var fshader = createShader(ctx, "FRAGMENT_SHADER", shader.fragment);
+
+  var program = ctx.createProgram();
+
+  if (!program) {
+    throw new Error("Failed to create program.");
+  }
+
+  ctx.attachShader(program, vshader);
+  ctx.attachShader(program, fshader);
+  ctx.linkProgram(program);
+
+  var linked = ctx.getProgramParameter(program, ctx.LINK_STATUS);
+  if (!linked) {
+    var lastError = ctx.getProgramInfoLog(program);
+    ctx.deleteProgram(program);
+    throw new Error("Error in program linking:" + lastError);
+  } else {
+    return program;
+  }
+}
+
+export const createImageShader = function (
+  ctx: WebGL2RenderingContext,
+  shader: Shader
+): ShaderProgram {
+  var program = createProgram(ctx, shader);
+  return {
+    program: program,
+    attr: [
+      {
+        bufferType: ctx.ARRAY_BUFFER,
+        buffer: ctx.createBuffer()!,
```

Comment:
@nswamy14  is there a possibility for createBuffer() to return null?

Reply:
Yes it can return null on certain cases.. like when error occurs.. etc